### PR TITLE
[TT-8093/TT-8579] update kin-openapi dependency to v0.115.0

### PR DESCRIPTION
**Pull request by jeffy-mathew**
## Description

Update [kin-openapi](https://github.com/getkin/kin-openapi) to latest version

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [x] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4947/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4655864816)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4947/merge
Commit: 139fb68a7297823eeda9de91931515d0a799ed41 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4655961267)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4947/merge
Commit: 9e4c3b7455ba7037a0d1a4d64a80a90090798ea7 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4655976918)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4947/merge
Commit: 839f66fcafcdc210cdd0f5dbc3043d4565ee7ea1 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4657157703)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4947/merge
Commit: 983f1d333bcb5c10e59e118ca9e1254fca8d23a6 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4675754886)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4947/merge
Commit: 62f7f4f1e85ea42918d4f49e71ffd69e6e476c0a 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4698204782)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4947/merge
Commit: 3b03137a0a5705be634d5d6cae1d04b2a906cb91 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4698416247)

File: apidef/oas/example_test.go [hunk 1]
```diff
@@ -93,7 +97,7 @@ func Test_exampleExtractor(t *testing.T) {
 			`{
 				"type": "string",
 				"example": "bird",
-				"enum": ["duck"]	
+				"enum": ["duck", "bird"]
```
Comment (furkansenharputlu) [inline apidef/oas/example_test.go:100]: I didn't get this why it changes here?

**Review by furkansenharputlu — APPROVED** [review]

File: apidef/oas/example_test.go (hunk 1 above)
Comment (jeffy-mathew) [reply apidef/oas/example_test.go:100]: example contains `bird` hence adding it to enum

**Pull request merged**